submission/proxy_server.go

```diff
@@ -43,8 +44,21 @@ func NewProxyServer(logListPath string, dBuilder DistributorBuilder, reqTimeout
 }
 
 // Run starts regular Log list updates. Blocks until initialization happens.
-func (s *ProxyServer) Run(logListRefreshInterval time.Duration, rootsRefreshInterval time.Duration) {
-	s.p.Run(context.Background(), logListRefreshInterval, rootsRefreshInterval)
+func (s *ProxyServer) Run(ctx context.Context, logListRefreshInterval time.Duration, rootsRefreshInterval time.Duration) {
+	s.p.Run(ctx, logListRefreshInterval, rootsRefreshInterval)
+
+	// start listening to proxy errs
+	go func() {
+		for {
+			select {
+			case <-ctx.Done():
```

Comment:
Can you confirm that context cancelation is the only way `s.p.Run` can terminate? If yes, please add the corresponding clarification to `Proxy.Run` comment.
If not, then this goroutine can hang forever, and needs fixing.

Reply:
Added comment.